### PR TITLE
Adds support for multiple $in, pointer / relation queries on $or

### PR DESCRIPTION
**Pull request by flovilmart**

### CODE REVIEW

File: src/Controllers/DatabaseController.js [hunk 1]
```diff
@@ -397,31 +397,33 @@ DatabaseController.prototype.owningIds = function(className, key, relatedIds) {
...
-        relatedIds = [query[key].objectId];
+  // Make it sequential for now, not sure of paralleization side effects
+  return Object.keys(query).reduce((promise, key) => {
+    return promise.then(() => {
+      if (query[key] &&
+          (query[key]['$in'] || query[key].__type == 'Pointer')) {
+        let t = schema.getExpectedType(className, key);
+        let match = t ? t.match(/^relation<(.*)>$/) : false;
+        if (!match) {
+          return;
```
Comment (nlutsenko) [inline src/Controllers/DatabaseController.js]: This should produce an inconsistent return type from the function, not sure if that is expected behavior or not. You are returning `void` or a `Promise` (later one down the line). Maybe convert to `Promise.resolve()`, but don't feel strongly.

Comment (flovilmart) [inline src/Controllers/DatabaseController.js]: np

Comment (nlutsenko) [inline src/Controllers/DatabaseController.js]: Still an inconsistent return type from a closure(lambda?), but nature of JS wins here.

**Comment by flovilmart** [comment]
Proposed fix for  #432 and #550

File: src/Controllers/DatabaseController.js [hunk 2]
```diff
@@ -397,36 +395,58 @@ DatabaseController.prototype.owningIds = function(className, key, relatedIds) {
...
+    let ors = query['$or'];
+    return Promise.all(ors.map((aQuery, index) => {
+      return this.reduceInRelation(className, aQuery, schema).then((aQuery) => {
+        query['$or'][index] = aQuery; 
+      })
+    }));
   }
-  return Promise.resolve();
+
+  return Object.keys(query).reduce((promise, key) => {
```
Comment (drew-gross) [inline src/Controllers/DatabaseController.js]: I'd suggest using `Promise.all

Comment (flovilmart) [inline src/Controllers/DatabaseController.js]: I was wondering about race conditions in that case

Comment (drew-gross) [inline src/Controllers/DatabaseController.js]: `all` can lead to race conditions, but I think in this case it won't.

**Comment by drew-gross** [comment]
Can you prefer to use promises in tests instead of the backbone style callbacks? We are moving away from the backbone style. No need to spend a lot of time fixing this one, just for next tests.

**Comment by flovilmart** [comment]
No problemo, I'll do the Promise.all and fix the tests.

**Pull request merged**